tests/UnixServerTest.php

```diff
@@ -0,0 +1,277 @@
+
+        $this->server->on('connection', function ($conn) use ($mock) {
+            $conn->on('end', $mock);
+        });
+        $this->loop->tick();
+        $this->loop->tick();
+    }
+
+    /**
+     * @covers React\EventLoop\StreamSelectLoop::tick
+     * @covers React\Socket\Connection::end
+     */
+    public function testConnectionDoesEndWhenClientCloses()
+    {
+        $client = stream_socket_client($this->uds);
+
+        fclose($client);
+
+        $mock = $this->expectCallableOnce();
+
+        $this->server->on('connection', function ($conn) use ($mock) {
+            $conn->on('end', $mock);
+        });
+        $this->loop->tick();
+        $this->loop->tick();
+    }
+
+    public function testCtorAddsResourceToLoop()
+    {
+        $loop = $this->getMockBuilder('React\EventLoop\LoopInterface')->getMock();
+        $loop->expects($this->once())->method('addReadStream');
+
+        $server = new UnixServer($this->getRandomSocketUri(), $loop);
+    }
+
+    public function testResumeWithoutPauseIsNoOp()
+    {
+        $loop = $this->getMockBuilder('React\EventLoop\LoopInterface')->getMock();
+        $loop->expects($this->once())->method('addReadStream');
+
+        $server = new UnixServer($this->getRandomSocketUri(), $loop);
+        $server->resume();
+    }
+
+    public function testPauseRemovesResourceFromLoop()
+    {
+        $loop = $this->getMockBuilder('React\EventLoop\LoopInterface')->getMock();
+        $loop->expects($this->once())->method('removeReadStream');
+
+        $server = new UnixServer($this->getRandomSocketUri(), $loop);
+        $server->pause();
+    }
+
+    public function testPauseAfterPauseIsNoOp()
+    {
+        $loop = $this->getMockBuilder('React\EventLoop\LoopInterface')->getMock();
+        $loop->expects($this->once())->method('removeReadStream');
+
+        $server = new UnixServer($this->getRandomSocketUri(), $loop);
+        $server->pause();
+        $server->pause();
+    }
+
+    public function testCloseRemovesResourceFromLoop()
+    {
+        $loop = $this->getMockBuilder('React\EventLoop\LoopInterface')->getMock();
+        $loop->expects($this->once())->method('removeReadStream');
+
+        $server = new UnixServer($this->getRandomSocketUri(), $loop);
+        $server->close();
+    }
+
+    /**
+     * @expectedException RuntimeException
+     */
+    public function testListenOnBusyPortThrows()
+    {
+        if (DIRECTORY_SEPARATOR === '\\') {
+            $this->markTestSkipped('Windows supports listening on same port multiple times');
+        }
+
+        $another = new UnixServer($this->uds, $this->loop);
+    }
+
+    /**
+     * @covers React\Socket\UnixServer::close
+     */
+    public function tearDown()
+    {
+        if ($this->server) {
+            $this->server->close();
+        }
+    }
+
+    private function getRandomSocketUri()
+    {
+        return "unix://" . sys_get_temp_dir() . DIRECTORY_SEPARATOR . uniqid(rand(), true) . '.sock';
```

Comment:
Just nitpicking here but why not go with `uniqid('reactphp-socket-tests-unix-server-', true)` to make it explicit when debugging that it's a socket for the `UnixServer` test?

Reply:
Turned out to not be a good idea, see https://github.com/reactphp/socket/pull/120/commits/999376385b95f3fd70bdc757cea006821e2dd020. I've added a check for the client as well.

Reply:
Interesting, what happened?

Reply:
Warning is shown (in the tests without @) that the path is truncated. My thinking was that we should rather be explicit with failure instead of the user potentially reusing an existing socket due to same name.